packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -344,6 +344,12 @@ class Navigator extends StatefulWidget {
   }
 
   /// The state from the closest instance of this class that encloses the given context.
+  ///
+  /// Typical usage is as follows:
+  ///
+  /// ```dart
+  /// NavigatorState navigator = Navigator.of(context);
```

Comment:
I added it for Theme but for Navigator it's not clear to me why it would be hard to get the right context. If you're pushing and popping you'll be in a "routes" map and so you'll already be in the context of a builder function.